### PR TITLE
[#593] - added illustrations for onboarding

### PR DESCRIPTION
**Pull request by gasstan**
This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules.
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [ ] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [x] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingView.kt [hunk 1]
```diff
@@ -175,106 +171,121 @@ fun OnboardingMainContent(
...
-        BottomNav(onboardingStage.getProgress(), onboardingState::goNext)
     }
 }
 
 @Composable
 private fun TopNavButtons(onboardingState: OnboardingState) {
-    Row {
-        val currentStage = onboardingState.current.collectAsState().value
+    val currentStage = onboardingState.current.collectAsState().value
+    if (currentStage == OnboardingStage.ShieldedByDefault) return
```
Comment (ccjernigan) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingView.kt:185]: Just a style preference, but I'd prefer to keep the curly braces for if statements like this.  Otherwise it is easy for human reviewers to miss the return statement.

Comment (ccjernigan) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingView.kt:185]: Separately, I'd like to confirm with @Geffenz that we really do want to eliminate the skip button on this first stage.  Because having skip on stage 2/3 doesn't seem that useful.

**Review by ccjernigan — COMMENTED** [review]
Adding the scroll is great, thanks for doing that.  It really helps make the app work better across devices that we might not test against.

In addition to the scroll, I think we need to adjust the size of the images based on screen size.  We'll have to figure out the best way to do that in Compose that's also somewhat multiplatform aware.

On a Pixel 2 sized device, our onboarding is over scrolling pretty badly.
Onboarding 2 - Light-en-US and Pixel 2 screen size is a reasonable size for us to support.

**Comment by gasstan** [comment]
> Adding the scroll is great, thanks for doing that. It really helps make the app work better across devices that we might not test against.
> 
> In addition to the scroll, I think we need to adjust the size of the images based on screen size. We'll have to figure out the best way to do that in Compose that's also somewhat multiplatform aware.
> 
> On a Pixel 2 sized device, our onboarding is over scrolling pretty badly. Onboarding 2 - Light-en-US and Pixel 2 screen size is a reasonable size for us to support.

i agree, i think we should also add background to the footer so that the content under the next button is invisible, also the top bar should not be there for the prod build that would also help with the available space, as for the image size...do you think that adding weight to layout components would help ?

**Comment by ccjernigan** [comment]
> > Adding the scroll is great, thanks for doing that. It really helps make the app work better across devices that we might not test against.
> > In addition to the scroll, I think we need to adjust the size of the images based on screen size. We'll have to figure out the best way to do that in Compose that's also somewhat multiplatform aware.
> > On a Pixel 2 sized device, our onboarding is over scrolling pretty badly. Onboarding 2 - Light-en-US and Pixel 2 screen size is a reasonable size for us to support.
> 
> i agree, i think we should also add background to the footer so that the content under the next button is invisible, also the top bar should not be there for the prod build that would also help with the available space, as for the image size...do you think that adding weight to layout components would help ?

For the top bar, I've also wondered whether we might be better served with single more standard Android top bar (it starts to become helpful when you see the overflow menu on the last screen).  If the design called for a bar with a standard Android back button, title, and then a skip button E.g. `<- Zcash        Skip` then that would improve the screen height.

So it seems like we might want to create three followup issues:
 - Figure out the top bar on Android's onboarding
 - Figure out design for background behind the bottom buttons
 - Figure out image scaling e.g. https://developer.android.com/jetpack/compose/layouts/adaptive

Do you want to file those followup issue?  Then we can merge this and keep moving forward.

**Comment by gasstan** [comment]
> * https://developer.android.com/jetpack/compose/layouts/adaptive

I've created followup issues, so i think we are ready to merge

File: ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingViewTest.kt [hunk 2]
```diff
@@ -193,8 +184,13 @@ class OnboardingViewTest : UiTestPrerequisites() {
     fun stage_1_skip() {
         val testSetup = newTestSetup(OnboardingStage.ShieldedByDefault)
 
-        val skipButton = composeTestRule.onNodeWithText(getStringResource(R.string.onboarding_skip))
-        skipButton.performClick()
+        composeTestRule.onNodeWithText(getStringResource(R.string.onboarding_next)).also {
```
Comment (ccjernigan) [inline ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/onboarding/view/OnboardingViewTest.kt:187]: Since the skip button is back, we should revert this change.

**Review by ccjernigan — APPROVED** [review]
